tensorcircuit/mpscircuit.py

```diff
@@ -618,20 +598,20 @@ def wavefunction_to_tensors(
         :rtype: List[Tensor]
         """
         wavefunction = backend.reshape(wavefunction, (-1, 1))
+        n_tensors = int(np.round(np.log(wavefunction.shape[0]) / np.log(dim_phys)))
         tensors: List[Tensor] = []
-        while True:  # not jittable
+        for _ in range(n_tensors):
             nright = wavefunction.shape[1]
             wavefunction = backend.reshape(wavefunction, (-1, nright * dim_phys))
             wavefunction, Q = split_tensor(
                 wavefunction,
                 center_left=True,
-                **rules,
+                split=split,
             )
             tensors.insert(0, backend.reshape(Q, (-1, dim_phys, nright)))
-            if wavefunction.shape == (1, 1):
-                break
+        assert wavefunction.shape == (1, 1)
         if not norm:
-            tensors[-1] *= wavefunction[0, 0]
+            tensors[0] *= wavefunction[0, 0]
```

Comment:
then why there is not a `/=`

Reply:
Because what I'm doing here is "unnormalize the MPS in case normalization is not required" instead of "normalize the MPS in case normalization in required"